File: 35. Search Insert Position/solution.java
class Solution {
  public int searchInsert(int[] nums, int target) {
    int low = 0;
    int high = nums.length - 1;
    int mid = low + ((high - low) / 2);
    while (low <= high){
      mid = low + ((high - low) / 2);
      int value = nums[mid];
      if (value == target){
        return mid;
      }
      else if (value < target){
        low = mid + 1;
      }
      else if (value > target){
        high = mid - 1;
      }
    }
    if (nums[mid] < target){
      return mid + 1;
    }
    return mid;
  }
}

File: 35. Search Insert Position/solution.py
class Solution(object):
  def searchInsert(self, nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: int
    """
    low = 0
    high = len(nums) - 1
    mid = low + ((high - low) / 2)
    while low <= high:
      mid = low + ((high - low) / 2)
      value = nums[mid]
      if value == target:
        return mid
      elif value < target:
        low = mid + 1
      elif value > target:
        high = mid - 1
    if nums[mid] < target:
      return mid + 1
    return mid